File: samples-and-tests/lists-with-gae/app/controllers/Application.java
package controllers;

import play.mvc.*;
import play.modules.gae.*;


public class Application extends Controller {

    public static void index() {
        if(GAE.isLoggedIn()) {
            Lists.index();
        }
        render();
    }
    
    public static void login() {
        GAE.login("Application.index");
    }
    
    public static void logout() {
        GAE.logout("Application.index");
    }

}

File: samples-and-tests/test-scala/app/controllers/Application.scala
package controllers

import play._
import play.mvc._

object Application extends Controller with Secure {
    
    def index(name: String = "Guest") = render(name)
    
}

trait Secure extends Controller {
    
    @Before
    def check {
        request.user match {
            case user: String => renderArgs += ("connected" -> user)
            case _ => unauthorized
        }
    }
    
}